boku7/BokuLoader
# Remove GetProcAddress + other stuff

Hello!

The most relevant part of this PR, is that the API `ntdll!GetProcAddress` is not longer used, we now find all the import addresses "by hand" with `xGetProcAddress`.

I found that the `NOHEADERCOPY` option was breaking Beacon for some reason, so what I did is to simply avoid writing the headers if that option was set (we no longer free the memory).  
Feel free to test this out and check if I am right!

I noticed that if you search the address of an API that does not exist, it breaks. That could actually happen if you tried to bypass ETW on an old system, where `ntdll!EtwEventWrite` would not exist.  
This is now fixed and we simply return  NULL and do nothing.

I specified the size of the UDRL to 100K because if you enable all features, the 64-bit loader exceeds the 5K boundary.

Also:
- we now check for NTSTATUS error codes
- removed many casts
- search DLLs with name in ASCII instead of wide string
- check machine type of rdll, so we don't inject a 32-bit PE into a 64-bit process or vise-versa
- when no syscalls are used, removed Virtual* APIs, use Nt* instead
- getSyscallNumber is now coded in asm
- some other very minor changes here and there

I think this is a very robust and reliable UDRL 😄 
Cheers!
